sklearn/utils/estimator_checks.py

```diff
@@ -340,6 +342,8 @@ def check_estimator_sparse_data(name, Estimator):
     X_csr = sparse.csr_matrix(X)
     y = (4 * rng.rand(40)).astype(np.int)
     for sparse_format in ['csr', 'csc', 'dok', 'lil', 'coo', 'dia', 'bsr']:
+        if name == 'SpectralCoclustering':
```

Comment:
Even here, similar to above scenario [this](https://github.com/maniteja123/scikit-learn/blob/bf459f445c43a663df565bc0aa237f57acfc7264/sklearn/cluster/bicluster.py#L300) call is failing. This is the stacktrace

```
sklearn/cluster/bicluster.py:298: RuntimeWarning: invalid value encountered in multiply
  z = np.vstack((row_diag[:, np.newaxis] * u,
Estimator SpectralCoclustering doesn't seem to fail gracefully on sparse data: it should raise a TypeError if sparse input is explicitly not supported.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sklearn/utils/estimator_checks.py", line 353, in check_estimator_sparse_data
    estimator.fit(X, y)
  File "sklearn/cluster/bicluster.py", line 136, in fit
    return self._fit(X)
  File "sklearn/cluster/bicluster.py", line 301, in _fit
    _, labels = self._k_means(z, self.n_clusters)
  File "sklearn/cluster/bicluster.py", line 189, in _k_means
    model.fit(data)
  File "sklearn/cluster/k_means_.py", line 880, in fit
    X = self._check_fit_data(X)
  File "sklearn/cluster/k_means_.py", line 855, in _check_fit_data
    X = check_array(X, accept_sparse='csr', dtype=[np.float64, np.float32])
  File "sklearn/utils/validation.py", line 407, in check_array
    _assert_all_finite(array)
  File "sklearn/utils/validation.py", line 58, in _assert_all_finite
    " or a value too large for %r." % X.dtype)
ValueError: Input contains NaN, infinity or a value too large for dtype('float64').
```